### ISSUE TITLE
[SECURITY] Signing keys in production?!?

### ISSUE THREAD

**Issue by matrixnotorg** [matrix-org/matrix.org#364]
I noticed in your blog post that you were talking about doing a postmortem and steps you need to take. As someone who is intimately familiar with your entire infrastructure, I thought I could help you out.

There I was, just going about my business, looking for ways I could get higher levels of access and explore your network more, when I stumbled across GPG keys that were used for signing your debian packages. It gave me many nefarious ideas. I would recommend that you don't keep any signing keys on production hosts, and instead do all of your signing in a secure environment.

**Comment by RRREEEEEEEE** [comment]
Lol jesus, they better hire an actual admin instead of soy devs
FYI https://www.name.com/domain/search/matrix.soy

**Comment by aveao** [comment]
> hire

AFAIK no one works on matrix full time, as a job.

> soy

Yikes. There's all these actual issues at hand (great work @matrixnotorg for uncovering all these issues, btw), and you go and attack them with a disproven scientific myth?

**Comment by RRREEEEEEEE** [comment]
@aveao What's "disproven" about soy? It's part of the Legume family and DOES exist, sorry if you don't want to believe in it.

**Comment by aveao** [comment]
@RRREEEEEEEE you know exactly what I mean.

Calling people "soy boys" or "soy devs" or whatever is bullshit and is based on the incorrect belief that  phytoestrogens in soy have a similar effect to regular estrogen (it does not on humans), and that this causes people to become "feminine" and "weak" (some even go as far to say that it causes someone to be a leftist, which doesn't make any sense).

Drop the topic.

**Comment by RRREEEEEEEE** [comment]
@aveao No, it's based on an ancient meme.
The actual fact is they have no actual admin, there's no myths here.

**Comment by RandomErrorMessage** [comment]
the threat of soy is real, just ask my brother's wife's boyfriend.

**Comment by samcday** [comment]
Is this Github or 4chan? I honestly can't tell right now.

**Comment by RRREEEEEEEE** [comment]
@aveao 
>AFAIK no one works on matrix full time, as a job.

Sorry I almost forgot
https://new-vector.workable.com/

**Comment by DexterHaslem** [comment]
hehe soy gotem!!! LOL!!!

**Comment by ekollof** [comment]
I guess they need some battle-hardened sysadmins. Don't let devs sysadmin, it ends in tears.

**Comment by m00dy** [comment]
> @RRREEEEEEEE you know exactly what I mean.
> 
> Calling people "soy boys" or "soy devs" or whatever is bullshit and is based on the incorrect belief that phytoestrogens in soy have a similar effect to regular estrogen (it does not on humans), and that this causes people to become "feminine" and "weak" (some even go as far to say that it causes someone to be a leftist, which doesn't make any sense).
> 
> Drop the topic, let the actual issues be handled, not your bullshit science.

aferim :)

**Comment by nukeop** [comment]
that soy meme must have hit close to home

**Comment by justjanne** [comment]
@aveao don’t let these 4chan user’s stupid arguments get to you, just ignore + block them. They’re not worth the time :)

**Comment by aveao** [comment]
Hm, interesting, the patreon hints otherwise.

Though, that does seem like it's for the french govt deployment, not for the matrix.org instance.

---

TBH, matrix protocol itself is problematic.

Perhaps a re-design on proto accounting for the fact that bad actors exist, giving sufficient control of various factors to homeserver admins so that they can block out spammers, abusers, creepos.

Also: I was told over a year ago that registers would be closed on matrix.org to drive decentralization, but that hasn't happened yet. Perhaps now is a good time to do so.

---

Anyways, locking this thread might be a wise solution. It derailed too much. My apologies.

@m00dy: derail sorununa yardimci olmuyorsun.

@justjanne: will do, thanks ^^.

**Comment by anzetal** [comment]
yikes. who let these alt right nazi incel into this repository. I am absolutely sick of how disgusting they are behaving. I need to take a break with my nintendo switch

**Comment by RRREEEEEEEE** [comment]
@aveao 
New Vector is the company that actively develops matrix synapse and HOPEFULLY SOME DAY matrix dendrite

I agree, try not to derail threads in the future and keep subjects on topic.

**Comment by ekollof** [comment]
This attack has nothing to do with the security of the Matrix protocol. There were oversights made on how to keep the production environment secure. The attacker left really good advice on how to fix things, of which the strongest advice was policy/best practice-based. I hope the matrix team implements them.

**Comment by schittler** [comment]
This sounds like a good time to make this thread contributor only, it seems to have ended up on /g/ or something.

**Comment by Reisyukaku** [comment]
Ave quit shit posting. This isn't the switch scene.

**Comment by RRREEEEEEEE** [comment]
Yea, if he's been around there for a month they (new vector) are very lucky he's a grey hat and a nice guy.

**Comment by iSasuke7** [comment]
@aveao 🍆?

**Comment by AlexanderBalson** [comment]
Posting in a legendary thread

**Comment by anko** [comment]
@matrixnotorg Thank you for your restraint, and for trying to get these issues addressed. Have you posted a full writeup of your method anywhere? It would make a good learning resource. (I can't quite piece it together from these separate issues and the details in [the Matrix.org blog post](https://matrix.org/blog/2019/04/11/security-incident/).)

**Comment by dylangerdaly** [comment]
Is there a chance the full db was pulled before this went live?

**Comment by RRREEEEEEEE** [comment]
@Reisyukaku 
> Ave quit shit posting. This isn't the switch scene.

Did you know this?

>OK you guys some of the posters were joking but this person actually runs a DNS service for Nintendo Switch.
>https://ave.zone

**Comment by DexterHaslem** [comment]
too bad it has nothing to do with matrix huh

**Comment by ekollof** [comment]
The attacker owned an old jenkins install, and worked from there because internally the security wasn't up to snuff. When I was doing pentesting back in the day, I saw that a lot. Just read through his issue-tickets, you'll get an idea on what he did and how he did it.

**Comment by RRREEEEEEEE** [comment]
@DexterHaslem if it had something to do with matrix spec / servers it wouldn't be so funny. But it's just bad administration so it really isn't much to see.

**Comment by RRREEEEEEEE** [comment]
Yes the entire DB can 100% have been pulled, he said he started the attack a month ago.

**Comment by noirscape** [comment]
Can I just remind everyone that this is an **issue tracker**, not a meme board/chat log?

Discussions should be more relevant to the issues at hand. Comments calling the developers names are not going to be conductive to resolving issues, they're pointless cruft. Keep them somewhere else.

**Comment by RRREEEEEEEE** [comment]
Your post is nothing but added spam

**Comment by ekollof** [comment]
> Yes the entire DB can 100% have been pulled, he said he started the attack a month ago.

Yep, change your password. On the other hand, the encryption keys for encrypted chat are safe. At least with the Riot.im client, they won't leave your device.

**Comment by x029a** [comment]
> @aveao No, it's based on an ancient meme.
> The actual fact is they have no actual admin, there's no myths here.
> 
> > trying to make everything political
> 
> absolutely pathetic. There are issues here, you need to stop making random shit up.

I'm dead. Absolute murder. The term 'Phytoestrogen' was used unironically in the face of a soy boy meme. I'm dead.

**Comment by dylangerdaly** [comment]
So any unencrypted channel, chat and bridge is now public knowledge.

**Comment by RRREEEEEEEE** [comment]
> So any unencrypted channel, chat and bridge is now public knowledge.

Yes, correct. That includes deleted messages since they stay in the database.
Better yet, nobody should be using matrix.org in the first place. Run your own instance, that's the entire point of how it works.

**Comment by rain-1** [comment]
How about a harvard nutritional science research publication: https://www.hsph.harvard.edu/nutritionsource/soy/

> The Takeaway: Soy is a unique food that is widely studied for its estrogenic and anti-estrogenic effects on the body. Studies may seem to present conflicting conclusions about soy, but this is largely due to the wide variation in how soy is studied. Results of recent population studies suggest that soy has either a beneficial or neutral effect on various health conditions. Soy is a nutrient-dense source of protein that can safely be consumed several times a week, and is likely to provide health benefits—especially when eaten as an alternative to red and processed meat.

**Comment by ekollof** [comment]
> So any unencrypted channel, chat and bridge is now public knowledge.

What's the problem? I kinda treat unencrypted chat as public knowledge anyway. 

(PS: Although the shitposting is fun, can we stop it? Some people are trying to have a serious discussion)

**Comment by RRREEEEEEEE** [comment]
Who gives a fuck about soy or the posts about it LOL? do you guys serious care?
Peppers are the best plant anyway, fuck everything else.

**Comment by dylangerdaly** [comment]
Even if you run your own instance, if you invite _anyone_ on the Matrix HS, it'd be federated and thus leaked.

**Comment by simontime** [comment]
Is any of this helpful to the topic at hand?
Stop shitposting and discuss the issue.

**Comment by SRGOM** [comment]
I would believe Harvard but I believed WaPo and NYT and there was no Russia Collusion.

What do?

REEEEEEEEEEEEEEEEEEEEE

**Comment by RRREEEEEEEE** [comment]
> Even if you run your own instance, if you invite _anyone_ on the Matrix HS, it'd be federated and thus leaked.

Yes, I'd suggest not interacting with matrix.org members at all.

**Comment by x029a** [comment]
> Who gives a fuck about soy or the posts about it LOL? do you guys serious care?
> Peppers are the best plant anyway, fuck everything else.

Phewwwwwww. They're still talking about soy. Someone quoted a wiki about soy. My sides have reached exited velocity.

**Comment by rain-1** [comment]
OK How about a accredited journal medical publication that's done an actual cross section study showing soy consumption has minimal effects of semen quality: https://www.ncbi.nlm.nih.gov/pubmed/11880595

**Comment by neilisfragile** [comment]
(deleted off topic spam)
